src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -4184,6 +4188,10 @@ internal bool CheckValEscape(SyntaxNode node, BoundExpression expr, uint escapeF
 
                     return true;
 
+                case BoundKind.ArrayOrSpanCollectionLiteralExpression:
+                case BoundKind.CollectionInitializerCollectionLiteralExpression:
+                    return true;
```

Comment:
Added a PROTOTYPE comment to `GetValEscape()` and `CheckValEscape()`. As I understand the proposal though, `span = [1, 2, 3]` should not result an error because the array would be allocated on the heap in that case.